engine/src/witness/btc/btc_deposits.rs

```diff
@@ -80,18 +80,28 @@ fn deposit_witnesses(
 	let mut deposit_witnesses = Vec::new();
 	for tx in txs {
 		let tx_hash = tx.txid().as_raw_hash().to_byte_array();
-		for (vout, tx_out) in (0..).zip(&tx.output) {
-			if tx_out.value > 0 {
+
+		if let Some(deposit) = (0..)
```

Comment:
A valid usecase would be this: You want to swap your bitcoin into DOT and ETH. You generate two deposit channels, and send funds to both in a single transaction. We should allow this, but it wouldn't work with the suggested change. We should only filter out outputs in the same TX to the same destination address